File: 22-jsOop/beans/Trial.js
class Trial{
    static PASS_MARK = 120;
    static MARK_MIN_VALUE =0;
    static MARK_MAX_VALUE =100;


    constructor(account, mark1, mark2) {
        this._account = account;
        if(mark1 > Trial.MARK_MAX_VALUE || mark1 < Trial.MARK_MIN_VALUE){
            throw new RangeError("Illegal argument in mark1");
        }
        this._mark1 = mark1;
        if(mark2 > Trial.MARK_MAX_VALUE || mark2 < Trial.MARK_MIN_VALUE){
            throw new RangeError("Illegal argument in mark2");
        }
        this._mark2 = mark2;
    }
    get account(){
        return this._account;
    }

    get mark1(){
        return this._mark1;
    }

    get mark2(){
        return this._mark2;
    }

    isPassed(){
        return this._mark1 + this._mark2 >= Trial.PASS_MARK;
    }

    get clearAllResults(){
        this._mark1 = 0;
        this._mark2 = 0;
        return this;
    }

    getCopy(){
        return new Trial(this._account, this._mark1, this._mark2);
    }

    fieldsToString(){
        return `${this._account};${this._mark1};${this._mark2}`;
    }

    toString(){
        return `${this.fieldsToString()};${this.isPassed()}`;
    }
}

module.exports = Trial;

File: 15-Java8/src/by/gsu/epamlab/beans/Trial.java
package by.gsu.epamlab.beans;

import static by.Constants.*;

public class Trial {
    private String name;
    private int mark1;
    private int mark2;
    protected static final int PASS_MARK = 120;

    public Trial() {
    }

    public Trial(String name, int mark1, int mark2) {
        this.name = name;
        setMark1(mark1);
        setMark2(mark2);
    }

    public int getMark1() {
        return mark1;
    }

    public void setMark1(int mark1) {
        if (mark1 > MARK_MAX_VALUE || mark1 < MARK_MIN_VALUE){
            throw new IllegalArgumentException(EXC_ILLEGAL_MARK1);
        }
        this.mark1 = mark1;
    }

    public int getMark2() {
        return mark2;
    }

    public void setMark2(int mark2) {
        if (mark2 > MARK_MAX_VALUE || mark2 < MARK_MIN_VALUE){
            throw new IllegalArgumentException(EXC_ILLEGAL_MARK2);
        }
        this.mark2 = mark2;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isPassed() {
        return mark1 + mark2 >= PASS_MARK;
    }

    public void clearAllResults(){
        mark1 = MARK_MIN_VALUE;
        mark2 = MARK_MIN_VALUE;
    }

    public Trial getCopy(){
        return new Trial(name, mark1, mark2);
    }

    protected String fieldsToString(){
        return name + DELIMITER + mark1 + DELIMITER + mark2;
    }

    @Override
    public String toString() {
        return fieldsToString() + DELIMITER + isPassed();
    }

}
